packages/squiggle-lang/src/rescript/Distributions/GenericDist.res

```diff
@@ -159,6 +119,89 @@ let toPointSet = (
   }
 }
 
+module Score = {
+  type scoreDistOrScalar = DistributionTypes.DistributionOperation.scoreDistOrScalar
+
+  type pointSet_ScoreDistOrScalar = D(PointSetTypes.pointSetDist) | S(float)
+  let argsMake = (
+    ~esti: scoreDistOrScalar,
+    ~answ: scoreDistOrScalar,
+    ~prior: option<scoreDistOrScalar>,
+  ): result<PointSetDist_Scoring.scoreArgs, error> => {
+    let toPointSetFn = toPointSet(
+      ~xyPointLength=MagicNumbers.Environment.defaultXYPointLength,
+      ~sampleCount=MagicNumbers.Environment.defaultSampleCount,
+      ~xSelection=#ByWeight,
+    )
+    let prior': option<result<pointSet_ScoreDistOrScalar, error>> = switch prior {
+    | None => None
+    | Some(Score_Dist(d)) => toPointSetFn(d, ())->E.R.bind(x => x->D->Ok)->Some
+    | Some(Score_Scalar(s)) => s->S->Ok->Some
+    }
+    let twoDists = (esti': t, answ': t): result<
+      (PointSetTypes.pointSetDist, PointSetTypes.pointSetDist),
+      error,
+    > => E.R.merge(toPointSetFn(esti', ()), toPointSetFn(answ', ()))
+    switch (esti, answ, prior') {
+    | (Score_Dist(esti'), Score_Dist(answ'), None) =>
```

Comment:
I think I recommended this before; I think this function should be split up into at least two functions. One should convert the inputs into valid types, the second function should take those valid types and do switch for them.

Reply:
I could see that. 

```rescript
let firstFunction = (estimate: 'a, answer: 'b, prior: option<'a>): abstractScoreArgs<'a, 'b> => ...
let secondFunction = (arg: abstractScoreArgs<'a, 'b>): scoreArgs => ... 
```